accounts/abi/argument.go

```diff
@@ -113,7 +113,10 @@ func (arguments Arguments) unpackTuple(v interface{}, marshalledValues []interfa
 	}
 	// If the output interface is a struct, make sure names don't collide
 	if kind == reflect.Struct {
-		if err := requireUniqueStructFieldNames(arguments); err != nil {
+		if err := requireUniqueAbiFieldNames(arguments); err != nil {
+			return err
+		}
+		if err := requireUniqueStructFieldNames(typ); err != nil {
```

Comment:
How come that when unpacking a tuple, you check both for `requireUniqueAbiFieldNames` and `requireUniqueStructFieldNames`, but not when unpacking atomic?

Reply:
Mmm... you're right. I also will add a test to cover this case.